### PR TITLE
mailsy: init at 5.0.0

### PR DESCRIPTION
**Pull request by 404Wolf**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Solves https://github.com/NixOS/nixpkgs/issues/331854

Not sure what to do for a test here. This is my first time contributing, I tried to follow the guides but I'm happy to fix up anything I inevitably messed up.

### CODE REVIEW

**Comment by 404Wolf** [comment]
I accidentally added other commits to this PR while rebasing to fix a patch for the build, I'll figure out how to fix that now. Sorry!

File: pkgs/by-name/ma/mailsy/package.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    hash = "sha256-RnOWvu023SOcN83xEEkYFwgDasOmkMwSzJ/QYjvTBDo=";
+  };
+
+  npmDepsHash = "sha256-ljmqNmLvRHPdsKyOdDfECBXHTIExM6nPZF45lqV+pDM=";
+
+  npmFlags = ["--legacy-peer-deps"];
+
+  dontNpmBuild = true;
+
+  patches = [./fix-file-lookup.patch];
```
Comment (eljamm) [inline pkgs/by-name/ma/mailsy/package.nix:23]: The patch is pretty simple, so we can just do this instead:

```suggestion
  postPatch = ''
    substituteInPlace utils/index.js \
      --replace-fail 'dirname, "../data/account.json"' 'process.cwd(), "account.json"' \
      --replace-fail 'dirname, "../data/email.html"' 'process.cwd() + "email.html"'
  '';
```

File: pkgs/by-name/ma/mailsy/package.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  src = fetchFromGitHub {
+    owner = "BalliAsghar";
+    repo = "Mailsy";
+    rev = version;
+    hash = "sha256-RnOWvu023SOcN83xEEkYFwgDasOmkMwSzJ/QYjvTBDo=";
+  };
+
+  npmDepsHash = "sha256-ljmqNmLvRHPdsKyOdDfECBXHTIExM6nPZF45lqV+pDM=";
+
+  npmFlags = ["--legacy-peer-deps"];
```
Comment (eljamm) [inline pkgs/by-name/ma/mailsy/package.nix:19]: The file doesn't seem to be properly formatted. You should format it with [nixfmt](https://github.com/NixOS/nixfmt)

File: pkgs/by-name/ma/mailsy/package.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{
+  buildNpmPackage,
+  fetchFromGitHub,
+  lib,
```
Comment (eljamm) [inline pkgs/by-name/ma/mailsy/package.nix:4]: This is more of a cosmetic thing (as far as I know), but `lib` is usually the first input of the derviation.

```suggestion
  lib,
  buildNpmPackage,
  fetchFromGitHub,
```

**Review by eljamm — CHANGES_REQUESTED** [review]

**Comment by eljamm** [comment]
Also, the commit message is missing a `:`

```diff
-  mailsy init at 5.0.0
+  mailsy: init at 5.0.0
```

**Comment by 404Wolf** [comment]
@eljamm Thank you very much for the feedback, I've gone ahead and made the suggested changes! :)

File: pkgs/by-name/ma/mailsy/package.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{ lib, buildNpmPackage, fetchFromGitHub, }:
```
Comment (eljamm) [inline pkgs/by-name/ma/mailsy/package.nix:1]: I think you used the wrong formatter. The nixfmt you should use is [nixfmt-rfc-style](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=nixfmt-rfc-style)

```suggestion
{
  lib,
  buildNpmPackage,
  fetchFromGitHub,
}:
```

**Comment by eljamm** [comment]
@404Wolf Thanks, can you also change the PR title to `mailsy: init at 5.0.0` as well? This gives better context to which version `mailsy` was initialized from in the PR list.

**Review by eljamm — APPROVED** [review]
This looks good to me.

@404Wolf Just don't forget to remove the commit to add yourself as a maintainer from your [other PR](https://github.com/NixOS/nixpkgs/pull/334700) if this one gets merged first (and vice versa), else you'd be adding yourself twice (if a merge conflict doesn't happen).

That said, welcome to nixpkgs :)

**Comment by 404Wolf** [comment]
Thank you! Your feedback and help has been really phenomenal

**Pull request merged**